XPNet.CLR/Scenery/XPlaneScenery.cs

```diff
@@ -56,6 +57,7 @@ unsafe void Enumerator(string inFilePath, void* inRef)
 	public interface IXPSceneryObject : IDisposable
 	{
 		void Draw(int lighting, int earthRelativ, XPLMDrawInfo_t[] drawInfos);
+		IXPInstance CreateInstance(IEnumerable<string> inDataRefs);
```

Comment:
Ok, I can do that. Just one thought, if we create the XPInstance via the scenery object, we would have the chance to keep track of the created instances and automatically dispose them with the object. That would speak for the current approach. Would that change your opinion?

Reply:
There is another problem here, where the XPSceneryObject.CreateInstance would be a much more elegant solution: The XPSceneryObject holds a private reference to the SceneryObject (m_objectRef). So, if we have a Instance.Create(xpSceneryObject, dataRefs) in the API, how do we get that reference to call XPLMCreateInstance(m_objectRef, ...)? Making the field that holds this reference internal instead does not work, because we expect the interface IXPSceneryObject in Instance.Create, I guess we still need to use the interface, and we probably do not want to make it public. If we would stay with XPSceneryObject.CreateInstance, we have this reference easily available. Any ideas?